pkg/controllers/machinesetsync/machineset_sync_controller_test.go

```diff
@@ -156,56 +155,280 @@ var _ = Describe("With a running MachineSetSync controller", func() {
+						))
+				})
+
+				It("should update the labels", func() {
+					Eventually(k.Object(mapiMachineSet), timeout).Should(
+						HaveField("Labels", Equal(map[string]string{"foo": "bar"})),
+					)
+				})
+			})
+
+			Context("when the conversion fails", func() {
+				BeforeEach(func() {
+					By("Creating the CAPI and MAPI MachineSets")
+					mapiMachineSet = mapiMachineSetBuilder.Build()
+					capiMachineSet = capiMachineSetBuilder.WithOwnerReferences(
+						[]metav1.OwnerReference{
+							{APIVersion: "FooAPIVersion", Kind: "FooKind", Name: "FooName", UID: "123"},
+						}).Build()
+
+					Expect(k8sClient.Create(ctx, capiMachineSet)).Should(Succeed())
+					Expect(k8sClient.Create(ctx, mapiMachineSet)).Should(Succeed())
+
+					By("Setting the AuthoritativeAPI to ClusterAPI")
+					Eventually(k.UpdateStatus(mapiMachineSet, func() {
+						mapiMachineSet.Status.AuthoritativeAPI = machinev1beta1.MachineAuthorityClusterAPI
+					})).Should(Succeed())
+				})
+
+				It("should update the synchronized condition on the MAPI MachineSet", func() {
+					Eventually(k.Object(mapiMachineSet), timeout).Should(
+						SatisfyAll(
+							HaveField("Status.Conditions", ContainElement(
+								SatisfyAll(
+									HaveField("Type", Equal(consts.SynchronizedCondition)),
+									HaveField("Status", Equal(corev1.ConditionFalse)),
+									HaveField("Severity", Equal(machinev1beta1.ConditionSeverityError)),
+									HaveField("Reason", Equal("FailedToConvertCAPIMachineSetToMAPI")),
+								))),
+						))
+
+				})
+			})
+
 		})
 
-		It("should update the replica count", func() {
-			Eventually(k.Object(mapiMachineSet), timeout).Should(
-				HaveField("Spec.Replicas", Equal(ptr.To(int32(4)))),
-			)
+		Context("when the conversion has warnings", func() {})
+
+		Context("when the CAPI MachineSet does not exist", func() {
+			BeforeEach(func() {
+				By("Creating the CAPI and MAPI MachineSets")
+				mapiMachineSet = mapiMachineSetBuilder.Build()
+				Expect(k8sClient.Create(ctx, mapiMachineSet)).Should(Succeed())
+
+				By("Setting the AuthoritativeAPI to ClusterAPI")
+				Eventually(k.UpdateStatus(mapiMachineSet, func() {
+					mapiMachineSet.Status.AuthoritativeAPI = machinev1beta1.MachineAuthorityClusterAPI
+				})).Should(Succeed())
+			})
+
+			It("should update the synchronized condition on the MAPI MachineSet", func() {
+				Eventually(k.Object(mapiMachineSet), timeout).Should(
+					SatisfyAll(
+						HaveField("Status.Conditions", ContainElement(
+							SatisfyAll(
+								HaveField("Type", Equal(consts.SynchronizedCondition)),
+								HaveField("Status", Equal(corev1.ConditionFalse)),
+								HaveField("Severity", Equal(machinev1beta1.ConditionSeverityError)),
+								HaveField("Reason", Equal("FailedToGetCAPIMachineSet")),
+								HaveField("Message", Equal("error synchronizing CAPI to MAPI, CAPI machineset not found")),
+							))),
+					))
+
+			})
+
```

Comment:
Left a comment on this, but this isn't actually the intended behaviour in this case, the sync should create the CAPI MachineSet based on the MAPI MachineSet